### DISCUSSION TITLE
Example usage of `Capture group search`  please

### DISCUSSION THREAD

**Discussion by fade2gray** [dnGrep/dnGrep#714] [Q&A] [answered]
I understand the usage of `Highlight groups`  in order to highlight capturing groups in the search results, but I can't figure out the purpose of `Capture group search` when using regular expressions for the file pattern match.

Edit: I  think I've figured it out.
With `Capture group search` enabled `(.*\.xaml)` will match `.xaml` files exclusively and ignore `.xaml.cs` files for example.

...but, I also can't figure the purpose of `Search parallel` 🤔

**Comment by doug24** [comment, marked as answer]
Capture Group Search is the answer to the question: how can I search for the filename or part of the filename inside the file?

Did you see this page?  https://github.com/dnGrep/dnGrep/wiki/Capture-Group-Search
Use a regular expression with a capture group in the file filter (Patterns to match).  Then reference that capture group in the Search for.

`Search parallel` runs file searches in multiple background threads.  When unchecked, only one thread is used (slower, but uses fewer resources; also useful for debugging).
